lib/rubocop/cop/performance/unneeded_sort.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop is used to identify instances of sorting and then taking
+      # only the first or last element.
+      #
+      # @example
+      #   # bad
+      #   [].sort.first
+      #   [].sort_by(&:length).last
+      #
+      #   # good
+      #   [].min
+      #   [].max_by(&:length)
+      class UnneededSort < Cop
+        include RangeHelp
+
+        MSG = 'Use `%<suggestion>s` instead of '\
+              '`%<sorter>s...%<accessor>s`.'.freeze
```

Comment:
What do you think of using the actual code snippet here instead of `...`?

Reply:
Sounds reasonable, but I'm not completely sold! I'm a little worried about the case when the actual code snippet is nontrivial:
```
Use `min_by` instead of `sort_by...first`
```
is clear about what both the offense and preferred approach are, while
```
Use `min_by do |x|
  a = x.first
  b = x.last
  [a.length, b.length].max
end` instead of `sort_by do |x|
  a = x.first
  b = x.last
  [a.length, b.length].max
end.first`
```
is more difficult to interpret. Any thoughts there?

If you do still want to do the the full code snippet approach, what's the best way to generate the preferred code outside of the `autocorrect` method? Is there a way to access a similar `Corrector` that doesn't actually affect the source code?

Reply:
Good point. I wasn't really thinking about multi-line examples. The shorthand using `sort_by...first` will be fine.